tests/src/3d/testqgs3drendering.cpp

```diff
@@ -486,56 +486,111 @@ void TestQgs3DRendering::testPhongShading()
 
 void TestQgs3DRendering::testExtrudedPolygonsDataDefined()
 {
-  QgsPropertyCollection propertyColection;
-  QgsProperty diffuseColor;
-  QgsProperty ambientColor;
-  QgsProperty specularColor;
-  diffuseColor.setExpressionString( QStringLiteral( "color_rgb( 120*(\"ogc_fid\"%3),125,0)" ) );
-  ambientColor.setExpressionString( QStringLiteral( "color_rgb( 120,(\"ogc_fid\"%2)*255,0)" ) );
-  specularColor.setExpressionString( QStringLiteral( "'yellow'" ) );
-  propertyColection.setProperty( QgsAbstractMaterialSettings::Property::Diffuse, diffuseColor );
-  propertyColection.setProperty( QgsAbstractMaterialSettings::Property::Ambient, ambientColor );
-  propertyColection.setProperty( QgsAbstractMaterialSettings::Property::Specular, specularColor );
-  QgsPhongMaterialSettings materialSettings;
-  materialSettings.setDataDefinedProperties( propertyColection );
-  materialSettings.setAmbient( Qt::red );
-  materialSettings.setShininess( 1 );
-  QgsPolygon3DSymbol *symbol3d = new QgsPolygon3DSymbol;
-  symbol3d->setMaterialSettings( materialSettings.clone() );
-  symbol3d->setExtrusionHeight( 10.f );
-  QgsVectorLayer3DRenderer *renderer3d = new QgsVectorLayer3DRenderer( symbol3d );
-  mLayerBuildings->setRenderer3D( renderer3d );
-
   const QgsRectangle fullExtent = mLayerDtm->extent();
-
-  Qgs3DMapSettings *map = new Qgs3DMapSettings;
-  map->setCrs( mProject->crs() );
-  map->setExtent( fullExtent );
-  map->setLayers( QList<QgsMapLayer *>() << mLayerBuildings << mLayerRgb );
   QgsPointLightSettings defaultLight;
   defaultLight.setIntensity( 0.5 );
   defaultLight.setPosition( QgsVector3D( 0, 1000, 0 ) );
-  map->setLightSources( { defaultLight.clone() } );
 
-  QgsFlatTerrainGenerator *flatTerrain = new QgsFlatTerrainGenerator;
-  flatTerrain->setCrs( map->crs() );
-  map->setTerrainGenerator( flatTerrain );
+  /*
+   * phong material
+   */
+  QgsPropertyCollection propertyColectionPhong;
```

Comment:
Can you make new test functions for this, instead of appending to the existing function? Sequential tests like this get annoying when they fail one-by-one and require multiple push/wait for CI result/update mask cycles.

Reply:
Done.